core/common/src/main/java/alluxio/metrics/MetricsSystem.java

```diff
@@ -196,9 +196,15 @@ private static String constructSourceName() {
       default:
         break;
     }
+    String hostName;
+    try {
+      hostName = NetworkAddressUtils.getLocalHostMetricName(sResolveTimeout);
+    } catch (RuntimeException e) {
+      hostName = "unhnown";
+      LOG.error("Can't find local host name", e.getMessage());
+    }
     AlluxioConfiguration conf = new InstancedConfiguration(ConfigurationUtils.defaults());
-    return sourceKey != null && conf.isSet(sourceKey)
-        ? conf.get(sourceKey) : NetworkAddressUtils.getLocalHostMetricName(sResolveTimeout);
+    return sourceKey != null && conf.isSet(sourceKey) ? conf.get(sourceKey) : hostName;
```

Comment:
Can you help modify the code so that we only call `NetworkAddressUtils.getLocalHostMetricName`  if the condition `sourceKey != null && conf.isSet(sourceKey)` doesn't satisfy?

Reply:
May be add a `if` statement won't looks like better?
if (sourceKey != null && conf.isSet(sourceKey) {
    return conf.get(sourceKey);
}

Reply:
It will look better and avoid calling `NetworkAddressUtils.getLocalHostMetricName`, saving one call.

Reply:
Ok, I'll take it